packages/mui-system/src/createStyled.js

```diff
@@ -245,3 +234,38 @@ export default function createStyled(input = {}) {
     return muiStyledResolver;
   };
 }
+
+export function internal_applyStyled(props, componentName, overridesResolver) {
+  const styles = [
+    applyThemeOverrides(props, componentName, overridesResolver),
+    applyThemeVariants(props, componentName),
+    // applySystemSx(props, componentName),
```

Comment:
I feel like we may need to enable this one as well, but I am ok with waiting for a request.

Reply:
The `sx` prop is more complicated than the rest because it needs to modify `props`, see [the note on slack](https://mui-org.slack.com/archives/C0170JAM7ML/p1730292828489019?thread_ts=1730183583.351549&cid=C0170JAM7ML).